scripts/functions/diagnostic_tests.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+function check_failure()
+{
+    # TODO, we will implement some sort of regexp check on the output
+    # for the things we see most common (gethostbyname, etc!)
+    printf "Checking error for known solutions...
+TODO: this function is not yet complete. Basically the idea will be to take your error
+and attempt to match it with issues commonly found. You will have to do this yourself
+for now... Go to the following address, and see if the error you are receiving matches
+anything found there:
+
+https://mooseframework.inl.gov/help/troubleshooting.html
```

Comment:
yeah dont print this yet. no need to tell them what the idea will be

Reply:
👍  None of the scripts will print this yet. I am not running `check_failure`. This is something I want to add someday.

Reply:
sounds good